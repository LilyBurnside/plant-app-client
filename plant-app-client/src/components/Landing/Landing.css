.landing {
  width: 100%;
}

.title {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #DBFFB7;
  border-radius: 15px;
  padding: 50px 10px 75px 10px;
  margin-top: 50px;
}

.lower {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: 80px;
}

button {
  color: #494949 !important;
  text-decoration: none;
  background: white;
  padding: 20px;
  border: 4px solid #494949 !important;
  margin-top: 50px;
  border-radius: 15px;
}

button:hover {
  background-color: #D58BFA;
  color: white !important;
  transition: .25s;
}

.title h1, .title p, .title a {
  background-color: #DBFFB7;
}

.title p {
  padding: 0px 40px;
}