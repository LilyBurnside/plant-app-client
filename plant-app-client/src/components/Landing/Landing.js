import React from 'react'
import { Link } from 'react-router-dom'
// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
// import TokenService from '../../services/token-service'
import './Landing.css'

export default class Landing extends React.Component {
  
  //Once I set up auth I can render login vs logout

  // renderLoginLink() {
  //   return(
  //     <div className='Landing not-logged-in'>
  //       <Link
  //         to='/login'>
  //         Log in
  //       </Link>
  //       <Link
  //         to='/register'>
  //           Register
  //       </Link>
  //     </div>
  //   )
  // }

  render() {
    return(
      <div className='Landing'>
        <section className='title'>
          <h1>The Plant Shelf</h1>
          <p>Find the right plant for you</p>
          <Link to='/quiz'>
            <button className='find-match'>Find my match ></button>
          </Link>
        </section>

        <section className='lower'>
          <Link to='/login'>
            <button className='login'>Login to show my plant shelf</button>
          </Link>
        </section>
      </div>
    )
  }
}