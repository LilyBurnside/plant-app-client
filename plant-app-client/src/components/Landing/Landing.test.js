import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom';
import Landing from './Landing';

describe('Landing Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected if not logged in', () => {
    const wrapper = shallow(<Landing />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});