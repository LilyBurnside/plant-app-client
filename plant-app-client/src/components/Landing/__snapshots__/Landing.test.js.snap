// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Component renders UI as expected if not logged in 1`] = `
<div
  className="Landing"
>
  <section
    className="title"
  >
    <h1>
      The Plant Shelf
    </h1>
    <p>
      Find the right plant for you and add it to your plant wishlist!
    </p>
    <Link
      to="/quiz"
    >
      <button
        className="find-match"
      >
        Find my match &gt;
      </button>
    </Link>
  </section>
  <section
    className="lower"
  >
    <Link
      to="/login"
    >
      <button
        className="login-landing-button"
      >
        Login to show my wishlist
      </button>
    </Link>
  </section>
</div>
`;
